packages/react/src/SelectPanel/SelectPanel.features.stories.tsx

```diff
@@ -560,3 +560,49 @@ export const AsyncFetch: StoryObj<SelectPanelProps> = {
     },
   },
 }
+
+export const WithOnCancel = () => {
+  const intialSelection = items.slice(1, 3)
+  const [selected, setSelected] = useState<ItemInput[]>(intialSelection)
+  const [filter, setFilter] = useState('')
+  const filteredItems = items.filter(
+    item =>
+      // design guidelines say to always show selected items in the list
+      selected.some(selectedItem => selectedItem.text === item.text) ||
+      // then filter the rest
+      item.text.toLowerCase().startsWith(filter.toLowerCase()),
+  )
+  // design guidelines say to sort selected items first
+  const selectedItemsSortedFirst = filteredItems.sort((a, b) => {
+    const aIsSelected = selected.some(selectedItem => selectedItem.text === a.text)
+    const bIsSelected = selected.some(selectedItem => selectedItem.text === b.text)
+    if (aIsSelected && !bIsSelected) return -1
+    if (!aIsSelected && bIsSelected) return 1
+    return 0
+  })
+  const [open, setOpen] = useState(false)
+
+  return (
+    <FormControl>
+      <FormControl.Label>Labels</FormControl.Label>
+      <SelectPanel
+        title="Select labels"
+        placeholder="Select labels"
+        subtitle="Use labels to organize issues and pull requests"
+        renderAnchor={({children, ...anchorProps}) => (
+          <Button trailingAction={TriangleDownIcon} {...anchorProps} aria-haspopup="dialog">
+            {children}
+          </Button>
+        )}
+        open={open}
+        onOpenChange={setOpen}
+        items={selectedItemsSortedFirst}
+        selected={selected}
+        onSelectedChange={setSelected}
+        onCancel={() => setSelected(intialSelection)}
```

Comment:
could we show a more "real-life" implementation of this? Right now it always goes back to the initial items on cancel rather than what was actually selected when the panel opened:

https://github.com/user-attachments/assets/10627930-ffe7-43f8-bde3-2d9fde61213b

Reply:
Fixed!